synapse/handlers/profile.py

```diff
@@ -368,6 +372,39 @@ def set_avatar_url(self, target_user, requester, new_avatar_url, by_admin=False)
                 400, "Avatar URL is too long (max %i)" % (MAX_AVATAR_URL_LEN, ),
             )
 
+        # Enforce a max avatar size if one is defined
+        if self.max_avatar_size or self.allowed_avatar_mimetypes:
+            # Download the desired media file (possibly from a remote resource)
+            try:
+                server_name, media_id = new_avatar_url.split("/")[-2:]
+            except ValueError:
+                raise SynapseError(400, "Invalid avatar URL '%s' supplied" %
+                                   new_avatar_url)
+
+            responder, media_info = (
+                yield self.media_repository.get_media_from_cache_or_remote(
```

Comment:
This can only be called for remote media?

Reply:
Seems like this might work for local media as well? https://github.com/matrix-org/synapse-dinsic/blob/d6f47d37830021e4725837fb8d5811549808d702/synapse/rest/media/v1/media_repository.py#L322-L330

Reply:
`store.get_cached_remote_media`  only looks in the `remote_media_cache` table, which does not include local media?

Reply:
I'm now using `get_local_media`.